reactos/dll/win32/browseui/basebarsite.cpp

```diff
@@ -472,10 +711,155 @@ LRESULT CBaseBarSite::OnNotify(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bH
     if (notifyHeader->hwndFrom == m_hWnd)
     {
     }
+    bHandled = FALSE; /* forward notification to parent */
+    return 0;
+}
+
+LRESULT CBaseBarSite::OnCommand(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    HRESULT                             hResult;
+    CComPtr<IDockingWindow>             parentSite;
+
+    if (HIWORD(wParam) == BN_CLICKED && LOWORD(wParam) == IDM_BASEBAR_CLOSE)
+    {
+        hResult = fDeskBarSite->QueryInterface(IID_PPV_ARG(IDockingWindow, &parentSite));
+        if (!SUCCEEDED(hResult))
+        {
+            return E_FAIL;
+        }
+        parentSite->ShowDW(FALSE);
+        bHandled = TRUE;
+    }
     return 0;
 }
 
-HRESULT CreateBaseBarSite(REFIID riid, void **ppv)
+LRESULT CBaseBarSite::OnCustomDraw(LPNMCUSTOMDRAW pnmcd)
+{
+    switch (pnmcd->dwDrawStage)
+    {
+        case CDDS_PREPAINT:
+        case CDDS_PREERASE:
+            return CDRF_NOTIFYITEMDRAW;
+        case CDDS_ITEMPREPAINT:
+            if (fVertical)
+            {
+                REBARBANDINFO info;
+                WCHAR wszTitle[MAX_PATH];
+                DWORD index;
+                RECT rt;
+                HFONT newFont, oldFont;
+
+                index = SendMessage(RB_IDTOINDEX, fCurrentActiveBar->fBandID , 0);
+                ZeroMemory(&info, sizeof(info));
+                ZeroMemory(wszTitle, sizeof(wszTitle));
+                DrawEdge(pnmcd->hdc, &pnmcd->rc, EDGE_ETCHED, BF_BOTTOM);
+                // We also resize our close button
+                ::SetWindowPos(toolbarWnd, HWND_TOP, pnmcd->rc.right - 22, 0, 20, 18, SWP_SHOWWINDOW);
```

Comment:
Can we maybe query these magic values?
`SystemParametersInfo(SPI_GETNONCLIENTMETRICS)`, or `GetThemePartSize`, `GetThemeInt` or `GetThemeMetric`